data/src/main/kotlin/com/vultisig/wallet/data/common/FileHelper.kt

```diff
@@ -212,3 +216,32 @@ fun Uri.fileName(context: Context): String {
 
 internal fun Bitmap.compressPng(stream: OutputStream) =
     compress(Bitmap.CompressFormat.PNG, 100, stream)
+
+
+suspend fun Uri.processZip(context: Context): List<AppZipEntry> = withContext(Dispatchers.IO) {
+    val entries = mutableListOf<AppZipEntry>()
+    context.contentResolver.openInputStream(this@processZip)?.use { inputStream ->
+        ZipInputStream(inputStream).use { zipInputStream ->
+            var zipEntry = zipInputStream.nextEntry
+            while (zipEntry != null) {
+                if (!zipEntry.isDirectory) {
+                    val entryName = zipEntry.name
+                    val ext = File(entryName).extension.lowercase()
+                    if (FILE_ALLOWED_EXTENSIONS.contains(ext)) {
+                        try {
+                            val fileContent = zipInputStream.readBytes().toString(Charsets.UTF_8)
+                            coroutineContext.ensureActive()
+                            entries.add(AppZipEntry(entryName, fileContent))
```

Comment:
Are we adding all zip entries @aminsato ? Can we have some identifier at least?

Reply:
@aminsato lets try to add marker file like .app_vultisig as metadata in the zip file, when creating it